packages/fuels/src/cli/commands/dev/startFuelCore.ts

```diff
@@ -0,0 +1,110 @@
+import type { ChildProcessWithoutNullStreams } from 'child_process';
+import { spawn } from 'child_process';
+import { mkdirSync, writeFileSync } from 'fs';
+import { dirname, join } from 'path';
+import { getPortPromise } from 'portfinder';
+import kill from 'tree-kill';
+
+import type { FuelsConfig } from '../../types';
+import { findPackageRoot } from '../../utils/findPackageRoot';
+import { log, loggingConfig } from '../../utils/logger';
+
+import { defaultChainConfig } from './defaultChainConfig';
+
+export const killNode =
+  (core: ChildProcessWithoutNullStreams, killFn: (pid: number) => void) => () => {
+    if (core.pid) {
+      killFn(Number(core.pid));
+    }
+  };
+
+export const createTempChainConfig = (coreDir: string) => {
+  const chainConfigPath = join(coreDir, 'chainConfig.json');
+  const chainConfigJson = JSON.stringify(defaultChainConfig, null, 2);
+  mkdirSync(dirname(chainConfigPath), { recursive: true });
+  writeFileSync(chainConfigPath, chainConfigJson);
+  return chainConfigPath;
+};
+
+export const startFuelCore = async (
+  config: FuelsConfig
+): Promise<{
+  bindIp: string;
+  accessIp: string;
+  port: number;
+  providerUrl: string;
+  chainConfig: string;
+  killChildProcess: () => void;
+}> => {
+  log('Starting node..');
+
+  const coreDir = join(config.basePath, '.fuels');
+
+  const bindIp = '0.0.0.0';
+  const accessIp = '127.0.0.1';
+
+  const chainConfig = config.chainConfig ?? createTempChainConfig(coreDir);
+  const port = config.fuelCorePort ?? (await getPortPromise({ port: 4000 }));
+
+  const providerUrl = `http://${accessIp}:${port}/graphql`;
+
+  // This is the private key of the `consensus.PoA.signing_key` in `defaultChainConfig.ts`.
+  // This key is responsible for validating the transactions.
+  const consensusKey = '0xa449b1ffee0e2205fa924c6740cc48b3b473aa28587df6dab12abc245d1f5298';
+
+  const flags = [
+    'run',
+    ['--ip', bindIp],
+    ['--port', port.toString()],
+    ['--db-path', coreDir],
+    ['--min-gas-price', '0'],
+    ['--poa-instant', 'true'],
+    ['--consensus-key', consensusKey],
+    ['--chain', chainConfig],
+    '--vm-backtrace',
+    '--utxo-validation',
+    '--manual_blocks_enabled',
+  ].flat();
+
+  return new Promise((resolve, reject) => {
+    const pkgRootDir = findPackageRoot();
+    const fuelsCorePath = join(pkgRootDir, 'node_modules', '.bin', 'fuels-core');
+
+    const command = config.useBuiltinFuelCore ? fuelsCorePath : 'fuel-core';
```

Comment:
```suggestion
    const command = config.useBuiltinFuelCore ? join(findPackageRoot(), 'node_modules', '.bin', 'fuels-core') : 'fuel-core';
```

The same logic as [here](https://github.com/FuelLabs/fuels-ts/pull/1291#discussion_r1367167177).

Reply:
https://github.com/FuelLabs/fuels-ts/commit/f737c93f5b0773e953fc1adc5908a8e12c5d99b9